assets/js/modules/adsense/index.js

```diff
@@ -84,6 +76,18 @@ domReady( () => {
 			settingsEditComponent: SettingsEdit,
 			settingsViewComponent: SettingsView,
 			setupComponent: SetupMain,
+			checkRequirements: () => {
+				const isAdBlockerActive = select( STORE_NAME ).isAdBlockerActive();
+				if ( ! isAdBlockerActive ) {
+					return;
+				}
+
+				throw {
+					code: ERROR_CODE_ADBLOCKER_ACTIVE,
+					message: __( 'Ad blocker detected, you need to disable it in order to set up AdSense.', 'google-site-kit' ),
+					data: null,
```

Comment:
Nit-pick, but the `data` property could also be omitted when `null`. It only matters when it contains something meaningful. We intentionally don't validate any specific schema (except possibly `code` and `message`) for these since they come from the WP REST API and `data` could be anything.

Reply:
I was going to leave data off, except we have this util function that implies that it's a necessary param. https://github.com/google/site-kit-wp/blob/develop/assets/js/util/errors.js#L40 Unless we're not considering this error obj to be a WPError.

Reply:
Good call, makes sense! 👍